website/docs/proposals/nvidia-dynamo-integration.md

```diff
@@ -0,0 +1,1694 @@
+
+- 14-category classification with ModernBERT
+- Highest accuracy for complex queries
+- **Latency**: Moderate (comprehensive analysis)
+- **Use case**: Comprehensive intent understanding
+
+**Signal Fusion Layer:**
+
+- **Policy-driven decision making**: Combines signals with configurable priority
+- **Routing logic**:
+  1. Check keyword rules first (fastest)
+  2. If no keyword match, check similarity rules
+  3. If no similarity match, use BERT classification (fallback)
+- **Confidence scoring**: Each signal provides confidence score
+- **Override mechanism**: High-confidence signals can override lower-priority signals
+- **Observability**: All signals logged for analysis
+
+**System Intelligence Benefits of Fusion Routing:**
+
+- **Accuracy-Efficiency Balance**: Dynamically selects routing strategy based on query complexity—fast path (keyword) for deterministic patterns achieves minimal latency, while deep analysis (BERT) for complex queries ensures maximum accuracy
+- **Adaptive Intelligence**: System automatically chooses the most efficient signal that meets accuracy requirements, avoiding unnecessary computation
+- **Flexibility**: Easy to add new routing rules without model retraining, enabling continuous system optimization
+- **Robustness**: Multiple signals provide redundancy and cross-validation, reducing misclassification risk and improving overall system reliability
+- **Holistic Optimization**: Considers both accuracy and efficiency in every routing decision, maximizing system-level intelligence
+
+### 2.3 Differentiation Analysis: Complementary Strengths
+
+The two systems operate at **different layers** of the inference stack with **minimal overlap**:
+
+#### Semantic Router: Request Intelligence Layer
+
+```
+User Query → [Semantic Understanding] → Model Selection → Request Enrichment
+```
+
+- **What:** Understands query semantics, intent, and safety
+- **Why:** Routes to the right model for the task
+- **When:** Before request reaches infrastructure
+- **Optimization:** Accuracy, cost, security
+
+#### Dynamo Router: Infrastructure Efficiency Layer
+
+```
+Enriched Request → [Worker Selection] → KV Cache Optimization → GPU Scheduling
+```
+
+- **What:** Optimizes worker selection and resource allocation
+- **Why:** Maximizes GPU utilization and minimizes latency
+- **When:** After model selection, during execution
+- **Optimization:** TTFT, ITL, throughput
+
+#### Integration Value Proposition
+
+| Dimension | Semantic Router Alone | Dynamo Router Alone | **Integrated System** |
+|-----------|----------------------|---------------------|----------------------|
+| **Model Selection** | ✅ Semantic accuracy (14 categories) | ❌ No model awareness | ✅ Best model for task |
+| **Worker Selection** | ❌ No worker awareness | ✅ KV cache optimization | ✅ Optimal worker for model |
+| **Prompt Engineering** | ✅ Domain-aware system prompts | ❌ No prompt optimization | ✅ Optimized CoT & MoE matching |
```

Comment:
potentially the system prompt injection could impact the prefix cache, we should also monitor that